specs/GetFavicon.md

```diff
@@ -0,0 +1,122 @@
+                                        (LPARAM)icon);
+                            return S_OK;
+                        })
+                        .Get());
+
+                return S_OK;
+            })
+            .Get(),
+        &m_faviconChangedToken));  
+```
+## .NET / WinRT Registering a listener for favicon changes
+```c#
+webView.CoreWebView2.FaviconChanged += (CoreWebView2 sender, Object arg) =>
+{
+    System.IO.Stream stream = new System.IO.MemoryStream();
+    await webView.CoreWebView2.GetFaviconAsync(
+        CoreWebView2FaviconImageFormat.Png,
+        stream);
+    // setting the window Icon to the bitmap
+    this.Icon = BitmapFrame.Create(stream); 
+
+};
+```
+# API Notes
+If a Web page does not have a Favicon, then the event is not fired.
+See [API Details](#api-details) Section below for API reference
+# API Details
+## Win32 C++
+```cpp
+/// Interface for the Favicon changed event handler
+[uuid(A0CDE626-8E5E-4EE2-9766-58F3696978FE), object, pointer_default(unique)]
+interface ICoreWebView2FaviconChangedEventHandler : IUnknown {
+  /// Called to provide the implementer with the event args for the
+  /// corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] IUnknown* args);
+}
+
+[uuid(93ACC5AD-DC22-419E-9A3F-75D96A1538E4), object, pointer_default(unique)]
+interface ICoreWebView2GetFaviconCompletedHandler : IUnknown {
+  /// Called to provide the implementer with the event args for the
+  /// corresponding event.
+  HRESULT Invoke([in] HRESULT error_code);
+}
+
+[uuid(DC838C64-F64B-4DC7-98EC-0992108E2157), object, pointer_default(unique)]
+interface ICoreWebView2_10 : ICoreWebView2_9 {
+  /// Add an event handler for the `FaviconChanged` event.
+  /// `FaviconChanged` runs when the WebView favicon changes
+  HRESULT add_FaviconChanged(
+        [in] ICoreWebView2FaviconChangedEventHandler* eventHandler,
+        [out] EventRegistrationToken* token);
+
+  /// Removing the event handler for `FaviconChanged` event
+  HRESULT remove_FaviconChanged(
+      [in] EventRegistrationToken token);
+
+/// Async function for getting the actual image data of the favicon
+ HRESULT GetFaviconAsync(
+        [in] COREWEBVIEW2_FAVICON_IMAGE_FORMAT format,
+        [in] IStream * imageStream,
+        [in] ICoreWebView2GetFaviconCompletedHandler* eventHandler
+ );
+
+  /// used to access the current value of the favicon
+  /// If a page has no favicon then this returns a nullptr
+  [propget] HRESULT FaviconUri([out, retval] LPWSTR* value);
+
+}
+[v1_enum]
+typedef enum COREWEBVIEW2_FAVICON_IMAGE_FORMAT {
+    /// Indicates that the function should return the favicon in PNG format
+    COREWEBVIEW2_CAPTURE_PREVIEW_IMAGE_FORMAT_PNG,
+
+    /// Indicates that the function should return the favicon in JPG format
+    COREWEBVIEW2_CAPTURE_PREVIEW_IMAGE_FORMAT_JPEG,
+}
+```
+
+## .Net/ WinRT
+```c#
+[interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2")]
```

Comment:
We still want the namespace block and the new enum and everything. I just meant within the runtime CoreWebView2 block to have the interface_name block. Sorry I wasnt clear.